app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  validates :name,  presence: true, length: {maximum: Settings.users.max_name}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: Settings.users.max_pass},
+    format: {with: VALID_EMAIL_REGEX},uniqueness: {case_sensitive: false}
+  private
+    def downcase_email
+      self.email = email.downcase
```

Comment:
để là email.downcase! nhé, và cho a biết sự khác nhau giữa 2 method này

Reply:
email.downcase: trả về xâu kí tự viết thường
email.downcase!: trả về xâu kí tự viết thường nhưng sẽ trả về nil nếu xâu kí tự không có sự thay đổi. 
Vì case_sensitive: false nên nếu khi nhập email là 1 chuỗi viết thường,kết quả sẽ trả về của email.downcase sẽ được 1 xâu kí tự giống xâu lúc đầu=>fail; dùng email.downcase! dù nhập 1 xâu toàn chữ thường sẽ trả về nil=>khác chuỗi ban đầu=>pass